src/hooks/events.ts

```diff
@@ -0,0 +1,42 @@
+import { useMemo } from "react";
+
+import { useSelector } from "./useSelector";
+import { useVenueId } from "./useVenueId";
+import { useFirestoreConnect, isLoaded } from "./useFirestoreConnect";
+import { useSovereignVenueId } from "./useSovereignVenueId";
+
+export const useConnectWorldEvents = () => {
+  const venueId = useVenueId();
+
+  const { sovereignVenueId, isSovereignVenueIdLoading } = useSovereignVenueId(
+    venueId
+  );
+
+  useFirestoreConnect(() => {
+    if (isSovereignVenueIdLoading || !sovereignVenueId) return [];
+
+    return [
+      {
+        collectionGroup: "events",
+        where: [["worldId", "==", sovereignVenueId]],
+        storeAs: "worldEvents",
+      },
+    ];
+  });
```

Comment:
> - `useRelatedVenues(venueId: string)`
>   - This would 'exhaustively walk the full tree' to get all related venues (grandparents, parents, siblings, children,  grandchildren, etc) for the specified `venueId`, and probably only store/return their `venueIds`
>   - We can probably leverage/reuse aspects of `fetchSovereignVenueId` / `useSovereignVenueId` here (if not use them directly) (and likely can fix the 'infinite recursion' bug at the same time)

@mike-lvov I'm going to work on the logic for this in a standalone engineering script (that we can reuse to make the hooks/etc later if we choose to go the way i've proposed/detailed above; or can just use to identify related venues for data backfilling purposes if we choose to go a different way with this)

Reply:
Note: I have captured all of the above in https://github.com/sparkletown/internal-sparkle-issues/issues/699 (specifically https://github.com/sparkletown/internal-sparkle-issues/issues/699#issuecomment-834003796) for future reference/posterity.